pkgs/applications/version-management/gitlab/default.nix

```diff
@@ -4,11 +4,24 @@
 }:
 
 let
-  rubyEnv = bundlerEnv {
+  rubyEnv = bundlerEnv rec {
     name = "gitlab-env-${version}";
     inherit ruby;
     gemdir = ./rubyEnv- + "${if gitlabEnterprise then "ee" else "ce"}";
+    gemset =
+      let x = import "${gemdir}/gemset.nix";
```

Comment:
```suggestion
      let x = import (gemdir + "/gemset.nix");
```

Reply:
This should pass strict eval.

Reply:
Sorry for maybe offtopic, but do you know why? I don't understand it.

Reply:
When `gemdir` is used inside the string interpolation, the directory gets copied into the nix store.
If we then try to import `gemset.nix` from there, that's an import from derivation (IFD).

By assembling the path outside a string interpolation, we avoid that (because it stays just a path until `x` is used).

Reply:
I would think the derivation is still in the "string context" of `gemdir + "/gemset.nix` so this is still IFD and still prohibited by strict mode.

Reply:
@Ericson2314

All I can say is that

```
HOME=/homeless-shelter NIX_PATH=nixpkgs=$(pwd) nix-instantiate ./pkgs/top-level/release.nix -A gitlab --option restrict-eval true --option build-timeout 1800 --argstr system thesystem --show-trace
```

does work with the suggested change, but doesn't with what's currently in the PR ;-)

Reply:
Oh nevermind I thought we are computing something.